Demo/src/main/java/com/braintreepayments/demo/GooglePayFragment.java

```diff
@@ -52,38 +57,28 @@ public void onResume() {
                 return;
             }
 
-            if (GooglePayCapabilities.isGooglePayEnabled(getActivity(), configuration)) {
-
-                googlePayClient.isReadyToPay(getActivity(), (isReadyToPay, e) -> {
+            if (GooglePayCapabilities.isGooglePayEnabled(requireActivity(), configuration)) {
+                googlePayClient.isReadyToPay(requireActivity(), (isReadyToPay, e) -> {
                     if (isReadyToPay) {
                         googlePayButton.setVisibility(View.VISIBLE);
                     } else {
-                        showDialog("Google Payments are not available. The following issues could be the cause:\n\n" +
+                        String message = "Google Payments are not available. The following issues could be the cause:\n\n" +
                                 "No user is logged in to the device.\n\n" +
-                                "Google Play Services is missing or out of date.");
+                                "Google Play Services is missing or out of date.";
+                        alertPresenter.showDialog(this, message);
                     }
                 });
             } else {
-                showDialog("Google Payments are not available. The following issues could be the cause:\n\n" +
+                String message = "Google Payments are not available. The following issues could be the cause:\n\n" +
                         "Google Payments are not enabled for the current merchant.\n\n" +
-                        "Google Play Services is missing or out of date.");
+                        "Google Play Services is missing or out of date.";
+                alertPresenter.showDialog(this, message);
             }
         });
     }
 
-    @Override
-    public void onPaymentMethodNonceCreated(PaymentMethodNonce paymentMethodNonce) {
-        super.onPaymentMethodNonceCreated(paymentMethodNonce);
-
-        GooglePayFragmentDirections.ActionGooglePayFragmentToDisplayNonceFragment action =
-            GooglePayFragmentDirections.actionGooglePayFragmentToDisplayNonceFragment(paymentMethodNonce);
-
-        NavHostFragment.findNavController(this).navigate(action);
-    }
-
     public void launchGooglePay(View v) {
-        FragmentActivity activity = getActivity();
-        activity.setProgressBarIndeterminateVisibility(true);
+        FragmentActivity activity = requireActivity();
```

Comment:
Can we remove this and just pass `getActivity()` to the `requestPayment` method?